### PR TITLE
Command line tools

### PR DESCRIPTION
**Pull request by patriya-piyawiroj**

### CODE REVIEW

File: a.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+DEFINE_string(user, "", "Logs in as the given username");
+DEFINE_string(warble, "", "Creates a new warble with the given text");
+DEFINE_string(reply, "", "Indicates that the new warble is a reply to the given id");
+DEFINE_string(follow, "", "Starts following the given username");
+DEFINE_string(read, "", "Reads the warble thread starting at the given id");
+DEFINE_bool(profile, false, "Get the user's profile of following and followers");
+
+// Prints error message
+void PrintError(char* errorMessage, char* fields) {
+	//printf("\nError: %s, %s", errorMessage, fields);
```
Comment (barath) [inline a.cpp]: Remove commented code and add TODO.

File: a.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+DEFINE_string(registeruser, "", "Registers the given username");
+DEFINE_string(user, "", "Logs in as the given username");
+DEFINE_string(warble, "", "Creates a new warble with the given text");
+DEFINE_string(reply, "", "Indicates that the new warble is a reply to the given id");
+DEFINE_string(follow, "", "Starts following the given username");
+DEFINE_string(read, "", "Reads the warble thread starting at the given id");
+DEFINE_bool(profile, false, "Get the user's profile of following and followers");
+
+// Prints error message
+void PrintError(char* errorMessage, char* fields) {
```
Comment (barath) [inline a.cpp]: Better to change signature to take in const reference std::strings

File: a.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	//printf("\nError: %s, %s", errorMessage, fields);
+}
+
+// Registers the given non-blank username
+void RegisterUser(std::string username) {
+
```
Comment (barath) [inline a.cpp:23]: Add TODOs to any un-implemented methods.

File: a.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	//printf("\nError: %s, %s", errorMessage, fields);
+}
+
+// Registers the given non-blank username
+void RegisterUser(std::string username) {
+	
+}
+
+// Posts a new warble (optionally as a reply), returns the id of new warble
+std::string Warble(std::string username, std::string text, std::string warbleID="") {
```
Comment (barath) [inline a.cpp]: Take const reference arguments for non-primitives that won't be modified.

Also, add spaces around `=` in default arg.

File: a.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	return "";
```
Comment (barath) [inline a.cpp]: Fix indentation here and elsewhere.

File: a.cpp [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+// Returns this user's following and followers
+void Profile() {
+
+}
+
+int main(int argc, char* argv[]) {
+	google::ParseCommandLineFlags(&argc, &argv, true);
+
+	if (FLAGS_registeruser != "") {
```
Comment (barath) [inline a.cpp]: Minor: I tend to prefer this form as canonical for testing a string `foo`:

`if (!foo.empty())`

File: a.cpp [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+		return;
+	}
+
+	// if not user logged in, can not perform other functions
+	if (FLAGS_user == "") { 
+		//PrintError(MISSING_ARGUMENT_ERROR, "user");
+		return;
+	}
+
+	if (FLAGS_warble != "") {Warble(FLAGS_user, FLAGS_warble);}
```
Comment (barath) [inline a.cpp]: Move body of `if` to new line for each of these.

File: a.cpp (hunk 1 above)
Comment (patriya-piyawiroj) [reply a.cpp]: OK

File: a.cpp (hunk 3 above)
Comment (patriya-piyawiroj) [reply a.cpp:23]: OK

File: a.cpp (hunk 2 above)
Comment (patriya-piyawiroj) [reply a.cpp]: OK

File: a.cpp (hunk 4 above)
Comment (patriya-piyawiroj) [reply a.cpp]: Done

File: a.cpp (hunk 5 above)
Comment (patriya-piyawiroj) [reply a.cpp]: OK

File: a.cpp (hunk 6 above)
Comment (patriya-piyawiroj) [reply a.cpp]: Sure. Looks better.

File: a.cpp (hunk 7 above)
Comment (patriya-piyawiroj) [reply a.cpp]: Fixed.

**Comment by barath** [comment]
Thanks for making these changes.  Just for future reference, it's best to go through a second round of reviews before getting the go-ahead to merge.

**Pull request merged**